### ISSUE TITLE
Sharing to Reminders uses a weird title sometimes

### ISSUE THREAD

**Issue by DavidAnson** [Ranchero-Software/NetNewsWire#1957]
1. Hit Share from article
2. Choose Reminders.app
3. Hit Add button
4. Later, look in Reminders

Title of reminder is "See first unread article", Notes has the actual title of the article you shared, URLis correct.

I can't make this happen on demand from my iPad as I type this, but it seems to happen much (maybe most?) of the time on my iPhone when saving articles to read later.

**Comment by vincode-io** [comment]
I'm not sure I understand the problem.  Could you attach some screenshots that show the behavior you see as incorrect?

**Comment by DavidAnson** [comment]
Turns out I can’t make it happen from iPhone, either. But it comes up every few days, so I’ll include an image next time I see this.

One thing that would be interesting is whether the string “See first unread article“ appears anywhere in the code. It seems like it ought to (vs. that coming from Reminders) and that may provide a hint about how to trigger this.

**Comment by DavidAnson** [comment]
Yep, here it is:
https://github.com/Ranchero-Software/NetNewsWire/blob/e8045b0e8bfeef69f76301086d70e466d4e810c6/Shared/Activity/ActivityManager.swift#L72

I’ll read through some of this code this evening to see if I can figure out what to do to make this happen reliably.

**Comment by vincode-io** [comment]
Are you using Siri Shortcuts for anything?

**Comment by DavidAnson** [comment]
Yes I do, but not with this app.

**Comment by vincode-io** [comment]
This is a pretty interesting bug.  It implies that there is some interaction happening between the share sheet dialog and donated user activities.

I wonder if we are triggering an issue with how Handoff uses the clipboard.  This is most likely an Apple bug, but we still try to work around those as much as we can.  If you can figure out how to consistently trigger this, we might be able do something about it.

**Comment by DavidAnson** [comment]
Thanks, I’ll let you know!

**Comment by DavidAnson** [comment]
It happened just now! I have an image, but the GitHub iOS client doesn’t seem to let me paste it. I’ll add it shortly. I was paying attention this time and can tell that the problem is present as soon as the share sheet comes up. Furthermore, if I cancel and share again - either within the share sheet or from the app itself - the same problem is present. I don’t know if it’s relevant, but this is the article it happened for: https://daringfireball.net/linked/2020/03/30/rene-solo. Switching to another article shares fine and switching back to the above one remains broken. Perhaps it’s something to do with the way Gruber links out his titles?

**Comment by DavidAnson** [comment]
Here is the image I refer to in my comment above:
![63B657CB-BCC2-4051-A972-52703F7DA336](https://user-images.githubusercontent.com/1828270/77955191-9f2d0a00-7284-11ea-9f26-44245b223e7e.png)

**Comment by DavidAnson** [comment]
In case it matters, I do all my article reading from the “All Unread“ view, pressing the “next article“ button in the middle of the bottom toolbar.

**Comment by vincode-io** [comment]
It doesn't look like you used the share button in the toolbar.  Just to be sure, you long pressed the youtube link and shared that using the context menu to Reminders right?

**Comment by DavidAnson** [comment]
While viewing the article, I am hitting the “box with an up arrow“ button that is at the right side of the bottom toolbar with five items. The problem is no longer reproducing for me with that article when I went back to the “Today“ section and tried again just now.

**Comment by vincode-io** [comment]
I'm able to replicate the bug now.  You have to have previously hit the next unread button for it to happen.

**Comment by vincode-io** [comment]
This one is so weird.  The last donated NSUserActivity always overrode what ever was suggested for the title of the shared item.  I removed the NSUserActivity donation for next unread.  It would be nice for Siri to suggest this to the user, but not at the price of the share sheet being broken.

I'm not sure we won't run into this problem in other places that we are donating NSUserActivites and the share sheet is available.  Anyway, I've cleared this one up and the fix will be in the next iOS  release (5.0.1).
